extensions/http/jersey/src/main/java/org/eclipse/dataspaceconnector/extension/jersey/JerseyExtension.java

```diff
@@ -42,11 +48,19 @@ public void initialize(ServiceExtensionContext context) {
 
         jerseyRestService = new JerseyRestService(jettyService, typeManager, configuration, monitor);
 
+        provider = new ResourceInterceptorProvider();
+        jerseyRestService.registerInstance(() -> new ResourceInterceptorBinder(provider));
```

Comment:
Is there a particular reason why the binder is passed through a `Supplier` and not directly?

Reply:
yes, because there must be a new instance registered for every resource, otherwise Jersey will vomit all over itself